connector/src/main/java/org/geysermc/connector/entity/ThrowableEntity.java

```diff
@@ -52,22 +65,68 @@ public ThrowableEntity(long entityId, long geyserId, EntityType entityType, Vect
      */
     @Override
     public void tick(GeyserSession session) {
-        super.moveRelative(session, motion.getX(), motion.getY(), motion.getZ(), rotation, onGround);
+        moveAbsoluteImmediate(session, position.add(motion), rotation, onGround, false);
         float drag = getDrag(session);
-        float gravity = getGravity();
+        float gravity = getGravity(session);
         motion = motion.mul(drag).down(gravity);
     }
 
     protected void moveAbsoluteImmediate(GeyserSession session, Vector3f position, Vector3f rotation, boolean isOnGround, boolean teleported) {
-        super.moveAbsolute(session, position, rotation, isOnGround, teleported);
+        MoveEntityDeltaPacket moveEntityDeltaPacket = new MoveEntityDeltaPacket();
+        moveEntityDeltaPacket.setRuntimeEntityId(geyserId);
+
+        if (isOnGround) {
+            moveEntityDeltaPacket.getFlags().add(MoveEntityDeltaPacket.Flag.ON_GROUND);
+        }
+        setOnGround(isOnGround);
+
+        if (teleported) {
+            moveEntityDeltaPacket.getFlags().add(MoveEntityDeltaPacket.Flag.TELEPORTING);
+        }
+
+        if (position.getX() != lastBedrockPosition.getX()) {
```

Comment:
What if you made `moveAbsoluteImmediate` take in a position delta? (It looks like in most cases that's what `motion` is for doing for us). That way you don't need to store a `lastBedrockPosition` and instead you can check `if (positionDelta.getX() != 0)`. 
I also don't quite see the need for `lastBedrockRotation` when the `rotation` variable does the same thing.

Reply:
It looks like I can do that, I just need to calculate the delta for cases like `moveRelative`.

For `lastBedrockRotation`, I was covering for classes that override `getBedrockRotation()`. I guess I can remove it since that isn't used in any of the projectiles.

I could also just revert this change, I'm pretty sure in my testing `MoveEntityDelta` didn't visibly improve anything. I kept it in because I assume sending fewer data would be better.

Reply:
Ah, OK. I experimented with the MoveEntityDeltaPacket, and the only issue I ran into was that positions/rotations could be desynced when logging in. I don't see that being an issue here, so feel free to keep this in.